From 3226380dbd3adcaa97afc5263784240acce71104 Mon Sep 17 00:00:00 2001
From: Gatien Chevallier <gatien.chevallier@foss.st.com>
Date: Mon, 22 Jul 2024 11:07:17 +0200
Subject: [PATCH 1038/1695] drivers: stm32_ltdc: fix uninitialized variable

Fix uninitialized variable that leads to random boot fail as virtual
address field in the structure can be initialized to a random value.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 9d527f589cafaf4e830ddfa61cf4dc73c29f4181 drivers: stm32_ltdc: fix uninitialized variable]
Change-Id: I5905b1d2513184586b2e480ef23dbefae27477b0
Signed-off-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/394495
Tested-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Gatien CHEVALLIER <gatien.chevallier@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
---
 core/drivers/stm32_ltdc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/drivers/stm32_ltdc.c b/core/drivers/stm32_ltdc.c
index 9d9189462..8b3991205 100644
--- a/core/drivers/stm32_ltdc.c
+++ b/core/drivers/stm32_ltdc.c
@@ -411,7 +411,7 @@ static TEE_Result stm32_ltdc_probe(const void *fdt, int node,
 	struct dt_node_info dt_info = { };
 	uint32_t hwid = 0;
 	TEE_Result ret = TEE_ERROR_GENERIC;
-	struct io_pa_va io_base;
+	struct io_pa_va io_base = { };
 
 	ldev = calloc(1, sizeof(*ldev));
 	if (!ldev)
-- 
2.47.2

